api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreProperties.java

```diff
@@ -65,4 +67,102 @@ public BiometricEngine getBiometricEngine() {
+            handler = getComponentByPropertyName(propertyName, PixErrorHandlingService.class);
+        }
+        return handler;
+    }
+    
+    public PdqErrorHandlingService getPdqErrorHandlingService() {
+        String propertyName = RegistrationCoreConstants.GP_MPI_PDQ_ERROR_HANDLER_IMPLEMENTATION;
+        PdqErrorHandlingService handler = null;
+        if (isPropertySet(propertyName)) {
+            handler = getComponentByPropertyName(propertyName, PdqErrorHandlingService.class);
+        }
+        return handler;
+    }
+    
+    /**
+     * Get the PDQ endpoint
+     * @return
+     */
+    public String getPdqEndpoint() {
+        return Context.getAdministrationService().getGlobalProperty(
+                RegistrationCoreConstants.GP_MPI_PDQ_ENDPOINT, "localhost");
+    }
+
+    /**
+     * Get the PDQ port
+     * @return
+     */
+    public Integer getPdqPort() {
+        return Integer.valueOf(Context.getAdministrationService().getGlobalProperty(
+                RegistrationCoreConstants.GP_MPI_PDQ_PORT, "8989"));
+    }
+
+    public String getPixEndpoint() {
+        return Context.getAdministrationService().getGlobalProperty(
+                RegistrationCoreConstants.GP_MPI_PIX_ENDPOINT, "localhost");
+    }
+
+    public Integer getPixPort() {
+        return Integer.valueOf(Context.getAdministrationService().getGlobalProperty(
+                RegistrationCoreConstants.GP_MPI_PIX_PORT, "3600"));
+    }
+
+    public String getMpiUsername() {
+        return Context.getAdministrationService().getGlobalProperty(
+                RegistrationCoreConstants.GP_MPI_ACCESS_USERNAME, "admin");
+    }
+
+    public String getMpiPassword() {
+        return Context.getAdministrationService().getGlobalProperty(
+                RegistrationCoreConstants.GP_MPI_ACCESS_PASSWORD, "admin");
+    }
+
+    public String getShrPatientRoot() {
+        return Context.getAdministrationService().getGlobalProperty(
+                RegistrationCoreConstants.GP_MPI_PATIENT_ROOT, "1.2.3.4.5.9");
+    }
+
+    public String getReceivingApplication() {
+        return Context.getAdministrationService().getGlobalProperty(
+                RegistrationCoreConstants.GP_MPI_RECEIVING_APPLICATION, "CR");
+    }
+
+    public String getReceivingFacility() {
+        return Context.getAdministrationService().getGlobalProperty(
+                RegistrationCoreConstants.GP_MPI_RECEIVING_FACILITY, "Clinic");
+    }
+
+    public String getSendingApplication() {
+        return Context.getAdministrationService().getGlobalProperty(
+                RegistrationCoreConstants.GP_MPI_SENDING_APPLICATION, "OPENMRS");
+    }
+
+    public String getSendingFacility() {
+        return Context.getAdministrationService().getGlobalProperty(
+                RegistrationCoreConstants.GP_MPI_SENDING_FACILITY, "LOCATION");
+    }
+
+    public String getUniversalIdType() {
+        return Context.getAdministrationService().getGlobalProperty(
```

Comment:
Is this the name of the global identifier domain in the API?

Reply:
it is the HL7 identifier _**Type**_ 
see http://hl7-definition.caristix.com:9010/HL7%20v2.5.1/Default.aspx?version=HL7%20v2.5.1&dataType=CX
and http://hl7-definition.caristix.com:9010/HL7%20v2.5.1/Default.aspx?version=HL7%20v2.5.1&table=0203
in this case it is the type for the global identifier domain in the MPI

Reply:
Then I'd think it should be a constant rather than a GP, in OpenMRS there is no metadata data that specifies what kind of identifier type it is, i.e. nothing says that this is a DL vs SSN and this is the purpose of this field? Is it required? If not just omit it otherwise treat all as patient identifiers or medical record numbers, so MR or PI look like great candidates but we have to pick one.

Reply:
We think it has to be A GP because it needs to be configurable depending on the MPI you use.

Request body 
MSH|^~\&|||||20180803155106||ADT^A08^ADT_A01|f73e094c-9519-44f8-b5d6-fda50adce1aa|P|2.3.1|||AL PID|||A100165^^^2.25.71280592878078638113873461180761116318&2.25.71280592878078638113873461180761116318&*PI*~3e4f11e0-974f-11e8-b8b1-000d3a1983eb^^^2.16.840.1.113883.4.357&2.16.840.1.113883.4.357&hl7||b3^b3^^^^^L|null, null^^^^^^U|20180207|F|||home^^^^^^L
(I bolded where it shows up in the message)

For example for Medic CR, it has to be ISO instead of PI, in the case of a national system you might want use a NI for the global identifier domain?